src/plugins/data/server/search/routes/bsearch.ts

```diff
@@ -34,6 +36,17 @@ export function registerBsearchRoute(
       onBatchItem: async ({ request: requestData, options }) => {
         const { executionContext, ...restOptions } = options || {};
 
+        apm.addLabels(
+          omitBy(
+            {
+              appId: executionContext?.name,
+              id: executionContext?.id,
+              page: executionContext?.page,
+            },
+            isUndefined
+          )
+        );
```

Comment:
I wonder if this could be run inside the `executionContextService.withContext` instead of relying on every plugin to call it.